multy_core/src/ethereum/ethereum_token.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2018 by Multy.io
+ * Licensed under Multy.io license.
+ *
+ * See LICENSE for details
+ */
+
+#ifndef MULTY_TRANSACTION_TOKEN_ETHEREUM_H
+#define MULTY_TRANSACTION_TOKEN_ETHEREUM_H
+
+#include "multy_core/src/u_ptr.h"
+
+#include <string>
+#include <memory>
+
+namespace multy_core
+{
+namespace internal
+{
+struct EthereumTransactionDestination;
+typedef std::unique_ptr<EthereumTransactionDestination> EthereumTransactionDestinationPtr;
+
+class EthereumSmartContractPayload
+{
+public:
+    virtual ~EthereumSmartContractPayload();
+
+    virtual BinaryDataPtr serialize(const EthereumTransactionDestination& destination) const = 0;
```

Comment:
Comment:
Serializes smart contract call as transaction payload. Uses receiver destination for getting address and token amount to transfer.